### ISSUE TITLE
"If you receive a score of 1 (using the pre-modification weights) you stop and the architect records how many candidates you required."

### ISSUE THREAD

**Issue by Kevin-Jin** [vighneshbirodkar/dating_game#3]
According to the specs, I think the score that the architect returns to the matchmaker should be the dot product of the person's weights with the matchmaker's candidate, BEFORE the person makes updates to those weights. Because score is calculated after the call to person.send_guess_and_get_update() in architect.py, I think the code is calculating the score with the post-modification weights and sending it to the matchmaker.

Might have to clear this up with the professor. If the pre-modification weights should be used to calculate score, then simply move

    person.send_guess_and_get_update(matchmaker.weight_guess)

after

    best_score = max(best_score, score)

In both instances in architect.py (lines 20 and 31).

**Comment by vighneshbirodkar** [comment]
You are correct. That's what I mentioned in class, but I realize my code doesn't do it. I have fixed the issue in the latest commit. Please check

**Comment by Kevin-Jin** [comment]
Resolved.

**Comment by vighneshbirodkar** [comment]
@Kevin-Jin Thanks a lot for following up and giving suggestions.

**Comment by vighneshbirodkar** [comment]
@Kevin-Jin I think the professor made it clear. My original implementation was correct
